### PR TITLE
[FEATURE] Integration with stripe to add payment method to farmer

### PR DESCRIPTION
**Pull request by xandergroeneveld**

### CODE REVIEW

File: app/assets/javascripts/farmers/payments.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      $(this).hide();
+    });
+
+    $("#cancel").on("click", function(e){
+      e.preventDefault();
+      $("#payment-form").toggle();
+      $("#add-payment").show();
+    });
+
+    var $form = $('#payment-form');
```
Comment (groeneveld91) [inline app/assets/javascripts/farmers/payments.js]: can move up to line 4 to reference in lines 6 and 12

File: app/controllers/farmers/payments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Farmers::PaymentsController < Farmers::BaseController
+  def show    
+    @stripe_publishable_key = Rails.application.config.stripe_publishable_key
```
Comment (groeneveld91) [inline app/controllers/farmers/payments_controller.rb]: is this more idiomatic than `ENV.fetch("STRIPE_PUBISHABLE_KEY")` ?

File: app/services/stripe_service.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+class StripeService
```
Comment (groeneveld91) [inline app/services/stripe_service.rb]: for the foreseeable future only farmers will be paying on the platform yeh? no payments from workers at all?

File: app/views/farmers/payments/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<% content_for :dashboard_header do %>
+  <%= render partial: "includes/dashboard/masthead", locals: {header: "Update Payment Information", icon: "payment", description: "Please provide valid credit card details to enable publishing of job advertisments", live_link: false } %>
+<%end%>
+
+<div id="payment-status" class="ui segment basic">
+  <div class="ui success message <%= @valid_payment_method ? '' : 'hidden' %>">
+    You're all set. We have a valid form of payment recorded for your account. If you'd like to change your payment method, click "Add Payment Method" below.
+  </div>
+  <div class="ui warning message <%= @valid_payment_method ? 'hidden' : '' %>">
+    We do no have a valid form of payment registered for your account. This is because you have either not added one yet our we have had a problem billing your previously registered card. To add a valid form of payment click "Add Payment Method" below.
```
Comment (groeneveld91) [inline app/views/farmers/payments/show.html.erb]: We do not*

File: app/models/farmer.rb [hunk 5]
```diff
@@ -15,6 +15,22 @@ class Farmer < ActiveRecord::Base
 
   attr_accessor :referred_by_token
 
+  def update_stripe_customer_source(token)
+    begin
+      customer = Stripe::Customer.create(
```
Comment (groeneveld91) [inline app/models/farmer.rb]: did you mean to create a stripe customer here? should we change `update_stripe_customer_source` to reflect something more along the lines of `create` stripe customer ?

File: app/controllers/farmers/payments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    @valid_payment_method = current_farmer.stripe_customer_id && !current_farmer.stripe_delinquent
```
Comment (groeneveld91) [inline app/controllers/farmers/payments_controller.rb]: depending on how many times this is used, should we consider creating `Farmer.valid_payment_method`

File: app/views/farmers/payments/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  <button id="cancel" class="ui primary button basic">Cancel</button>
+  <input type="submit" class="ui submit primary button" value="Update Payment Information">
+</form>
+
+<% content_for :javascripts do %>
+  <script>
+    window.stripe_publishable_key = "<%= @stripe_publishable_key %>";
+  </script>
+  <script type="text/javascript" src="https://js.stripe.com/v2/"></script>
```
Comment (groeneveld91) [inline app/views/farmers/payments/show.html.erb]: Is this initializing properly for you? I'm getting `Uncaught ReferenceError: Stripe is not defined` in `payments.js:3`

File: app/assets/javascripts/farmers/payments.js (hunk 1 above)
Comment (xandergroeneveld) [inline app/assets/javascripts/farmers/payments.js]: nice

File: app/services/stripe_service.rb (hunk 3 above)
Comment (xandergroeneveld) [inline app/services/stripe_service.rb]: yep

File: app/models/farmer.rb (hunk 5 above)
Comment (xandergroeneveld) [inline app/models/farmer.rb]: oh sorry this should have been removed, all logic in the StripeService

File: app/controllers/farmers/payments_controller.rb (hunk 6 above)
Comment (xandergroeneveld) [inline app/controllers/farmers/payments_controller.rb]: yeh good call

Comment (xandergroeneveld) [inline app/controllers/farmers/payments_controller.rb]: will add tests for that method too, much nicer

**Pull request closed without merging**